### ISSUE TITLE
Inconsistent violations for NeedBraces/allowSingleLineStatement

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#2291]
Code:
````
package test;

public class TestClass {
	public static void t() {
		int count = 0;
		
		for (;;) // line 7
			count++;
		
		for (String s: new String[]{""}) // line 10
			count++;
	}
}
````
Config:
````
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
	<property name="charset" value="ASCII" />
	<property name="severity" value="error" />
	<property name="fileExtensions" value="java" />

	<module name="TreeWalker">
		<module name="NeedBraces">
			<property name="allowSingleLineStatement" value="true" />
		</module>
	</module>
</module>
````
Output:
````
Starting audit...
\TestClass.java:7: error: 'for' construct must use '{}'s.
Audit done.
Checkstyle ends with 1 errors.
````

I was expecting both lines to be reported as errors, since they are identical for loops, but the for-each loop isn't reported.
I used CS 6.11.2.
